### DISCUSSION TITLE
Multiselect not working when used in relationship.

### DISCUSSION THREAD

**Discussion by anu-dev007** [filamentphp/filament#4308] [Help]
Hi, 

I have a Resource named WarehouseUser which has hasOne relationship with Warehouse Model. My form in WarehouseUser Resource is something like

```
public static function form(Form $form): Form
{
    return $form->schema([
        Fieldset::make('Warehouse Owner Details')
            ->schema([
                // WarehouseUser Model Fields
            ]),

        Fieldset::make('Warehouse Details')
            ->relationship('warehouse')
            ->schema([
                //Warehouse Model Fields
            ])
    ]);    
}
```

Everything till this is working absolutely fine. But Now I need to add a `many to many` relationship named categories for Warehouse ( Not WarehouseUser ). I have made a pivot table for this and relationship is working if I test it outside Filament.

But when I use `multiselect field` inside `warehouse relationship` something like

```
Fieldset::make('Warehouse Details')
    ->relationship('warehouse')
    ->schema([
        Forms\Components\MultiSelect::make('categories')
            ->relationship('categories', 'name'),
    ])
```

It shows Categories in dropdown which indicated that `many to many` relation is made fine. 

![image](https://user-images.githubusercontent.com/107536597/193394597-e4ab4190-0d62-4790-9a81-7748d23dbbab.png)

Now when I click on create it doesn't insert values in `category_warehouse` pivot table. 

Also, One more interesting thing about this is when I use same form on edit page, it works. It inserts data in  `category_warehouse` pivot table. 

Hope that make sense. 

Please help on how can I fix this. Any alternative that I can opt?

**Comment by Z3d0X** [comment]
In create, your "warehouse" relation doesn't exist yet, therefore it cannot create "warehouse.categories"...

Your structure seems a bit odd to me. From the given information I'd have a Warehouse Resource with BelongsTo WarehouseUser (Select/Fieldset) relationship and Warehouse Category BelongsToMany multiselect. Not sure if this would fit your use case.

**Reply by anu-dev007** [reply]
Hi @Z3d0X, thanks for your reply but `warehouse` relation does exist. Every other field than `multi select` works fine. Only this  `multi select` field doesn't work

**Reply by Z3d0X** [reply]
Not sure if you understand what I was trying to say. What I meant to say was that, the reason you are probably having issues is that you are trying have a relationship select within a relationship.

In this case Warehouse Category Multiselect inside a Fieldset that is scoped to "warehouse"  relation

**Reply by anu-dev007** [reply]
Yes, thats what I want to fix. Although, it is working on edit page.

**Reply by anu-dev007** [reply]
I have created Warehouse Resource with BelongsTo WarehouseUser but it's not working fine.I've created this relation in warehouse.php file.
```
  public function wuser(){
          return $this->belongsTo(WarehouseUser::class,'user_id');
  
      }

```

```
                Fieldset::make('Warehouse Owner Details')
                         ->relationship('wuser')
                         ->schema([
                                  Forms\Components\TextInput::make('name')
                                  ->maxLength(255),
                            ]);
```
```
                  Fieldset::make('Warehouse Details')->schema([
                        Forms\Components\MultiSelect::make('categories')->relationship('categories', 'name'),
                        Forms\Components\Select::make('state_head_id')->options(User::role('state-head')->pluck('name', 'id'))
                       ->label('State Head')->required(),
                 ]);
```
This error is coming when i creating record .
![image](https://user-images.githubusercontent.com/107536597/193510948-4dce40c3-335c-4b6e-aa73-3740437d5edc.png)
